src/v/kafka/server/handlers/configs/config_response_utils.cc

```diff
@@ -317,6 +317,20 @@ static inline void add_topic_config_if_requested(
     }
 }
 
+// Instantiate explicitly for unit testing
+using describe_int_t = decltype(&describe_as_string<int>);
+template void add_topic_config_if_requested<int, describe_int_t>(
+  const describe_configs_resource& resource,
+  describe_configs_result& result,
+  std::string_view default_name,
+  const int& default_value,
+  std::string_view override_name,
+  const std::optional<int>& overrides,
+  bool include_synonyms,
+  std::optional<ss::sstring> documentation,
+  describe_int_t&& describe_f,
+  bool hide_default_override = false);
```

Comment:
@oleiman I didn't move these to a separate namespace (eg. `kafka::test`) like you recommended in your example because I felt that that would lead to too verbose code.
```c++
template add_topic_config_if_requested(...){...}

namespace test {
  template add_topic_config_if_requested(...){
    ... // std::forward arguments to kafka::add_topic_config_if_requested
  }
  // + explicit instantiation
  
  // x 3 to expose both templated methods and the describe_as_string helper
}
```
Let me know what you think.

Reply:
Finally realized why this looks sort of odd to me - I think most of the symbols in `config_response_utils.cc` don't need to be in `namespace kafka` at all (could/should be in the anon namespace instead). 

If that's the case, then your explicit instantiations wouldn't be visible to the tests, and you would need to implement shim functions in a non-anon namespace (e.g. `kafka` or `kafka::test` or whatever).

That being said, I fear I've opened an annoying can of worms here - referring back to [noah's comment](https://github.com/redpanda-data/redpanda/pull/16922#discussion_r1517177341) it might be better to write this off as not worth the effort or better addressed in a follow-up.

Reply:
Yeah, that's true. I think this PR is already getting too large so let's leave that as a potential follow-up.

Reply:
Ack. Last request (assuming the diff is ready): could you please combine some of the commits in the stack to the extent that it makes sense? 

level of granularity is up to your judgement, but at minimum the commits that move big chunks of code can be combined to show just the final location in the tree. Saves a bunch of time in review